integration_tests/src/main/python/orc_cast_test.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2020-2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from asserts import assert_gpu_and_cpu_are_equal_collect, assert_gpu_and_cpu_error
+from data_gen import *
+from pyspark.sql.types import *
+from spark_session import with_cpu_session
+from orc_test import reader_opt_confs
+
+
+def create_orc(data_gen_list, data_path):
+    # generate ORC dataframe, and dump it to local file 'data_path'
+    with_cpu_session(
+        lambda spark: gen_df(spark, data_gen_list).write.orc(data_path)
+    )
+
+
+int_gens = [boolean_gen] + integral_gens
+@pytest.mark.parametrize('offset', list(range(1, len(int_gens))), ids=idfn)
+@pytest.mark.parametrize('reader_confs', reader_opt_confs, ids=idfn)
+@pytest.mark.parametrize('v1_enabled_list', ["", "orc"])
+def test_read_type_casting_integral(spark_tmp_path, offset, reader_confs, v1_enabled_list):
```

Comment:
nit: This is a very minor nit and I am okay with merging this in as is. 

This feels overly complicated. You write out the file with all of the possible types in it. But then when you read it in, you shift the schema N times so each type is read as each other type? Yes that works, but it is complicated enough that I don't know if there is a off by one error with the range used in offset. Also the name of the test does not make it clear what is being tested. I would much rather have int_gens be passed in as a read_type parameter.

```
@pytest.mark.parametrize('read_type', int_gens, ids=idfn)
...
re_gen_list = [('c'+ str(i), read_type) for i, gen in enumerate(int_gens)]
```